### PR TITLE
[3.11] gh-103971: Fix wrong line number generated for match-case blocks

### PR DESCRIPTION
**Pull request by gaogaotiantian**
The residue location issue would cause the `case` block setting the line number of `RETURN_VALUE` to the code block just parsed, which confuses debugger to believe the last line of the code block was executed even though it was not.

* Issue: gh-103971

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2023-04-28-18-57-13.gh-issue-103971.Q3U9lv.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix the wrong line number assigned for match-case blocks
```
Comment (brandtbucher) [inline Misc/NEWS.d/next/Core and Builtins/2023-04-28-18-57-13.gh-issue-103971.Q3U9lv.rst]:
```suggestion
Fix an issue where incorrect locations numbers could be assigned to code following ``case`` blocks.
```

**Review by brandtbucher — APPROVED** [review]
Thanks for the quick fix (and welcome to the compiler)!

**Comment by gaogaotiantian** [comment]
Thank you for the guidance!

**Comment by brandtbucher** [comment]
Can you open a PR with just the new test against `main`?

**Pull request merged**